Sources/SnapshotTesting/Snapshotting/NSImage.swift

```diff
@@ -20,9 +20,18 @@ extension Diffing where Value == NSImage {
       let message = new.size == old.size
         ? "Newly-taken snapshot does not match reference"
         : "Newly-taken snapshot@\(new.size) does not match match reference@\(old.size)"
+      let oldAttachment = XCTAttachment(image: old)
+      oldAttachment.name = "reference"
+      oldAttachment.lifetime = .keepAlways
```

Comment:
What the reason for this change? It seems like we'd only want to keep attachments around for failing tests and clean up after them when the tests pass, no?

Reply:
I'm not super precious about this so if we think it's best to not have it then that's cool too but my reasons were:

1. If we don't have this then it's not really possibly to have a unit test which demonstrates the usage.

2. If a snapshot has failed, I feel like we should still report the failure regardless and leave the reporting decisions up to the end client.

    • Perhaps one example would be if we run tests on 10 devices, and only one device fails we choose to have it show as passed but our custom reporting tool is able to still see the diffs and show that in a way which makes sense